### PR TITLE
TEST-#5341: Bump test-ray-master ray to 3.0.

### LINKED ISSUE #5341: TEST: CI: Bump test-ray-master branch version to latest release at 3.0

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5341 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: .github/workflows/push-to-master.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
           # for more info see https://github.com/conda-incubator/setup-miniconda/issues/264
           use-only-tar-bz2: false
       - name: install Ray nightly build
-        run: pip install https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp38-cp38-manylinux2014_x86_64.whl
+        run: pip install https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-3.0.0.dev0-cp38-cp38-manylinux2014_x86_64.whl
```
Comment (anmyachev) [inline .github/workflows/push-to-master.yml:27]: Have you tested it locally? (the installation itself, otherwise we won’t know until we merge it)

Comment (mvashishtha) [reply .github/workflows/push-to-master.yml:27]: Yep, I ran all the tests in this job and they passed.

Comment (anmyachev) [reply .github/workflows/push-to-master.yml:27]: Good, thanks!

**Review by anmyachev — APPROVED** [review]
LGTM

**Review by pyrito — APPROVED** [review]

**Pull request merged**